product/views.py

```diff
@@ -0,0 +1,101 @@
+import json
+from datetime            import datetime, timedelta
+
+from django.http         import JsonResponse
+from django.views        import View
+from django.db.models    import Q
+from django.utils        import timezone
+
+from .models import (
+        Product,
+        Category,
+        ProductImage,
+        ProductLike,
+        Review,
+        ReviewStatus,
+)
+
+def is_new(create_at, compare_date):
+    return True if create_at > compare_date else False
+
+def check_best_list():
+    best_query = Product.objects.all().order_by('total_sales')[:10]
+    return [best.id for best in best_query]
+
+def is_best(checkList, id):
+    return True if id in checkList else False 
+
+def is_sale(sale):
+    return True if sale > 0 else False
+
+class ProductListView(View):
+    def get(self, request):
+        category_name   = request.GET.get('category', None)
+        sorting         = request.GET.get('sorting', '-total_sales')
+        best_list       = check_best_list()
+        compare_date    = timezone.localtime() - timedelta(days=30)
+
+        product_list = Product.objects.all().order_by(sorting) \
+            if category_name is None else Product.objects.filter(category__name =\
+            category_name).order_by(sorting)
+
+        products = [
+               {
+                    'id'            : item.id,
+                    'name'          : item.name,
+                    'price'         : item.price,
+                    'sale'          : item.sale,
+                    'stock'         : item.stock,
+                    'imageUrl'      : item.image_url,
+                    'category'      : item.category.id,
+                    'isNew'         : is_new(item.create_at, compare_date),
+                    'isBest'        : is_best(best_list, item.id),
+                    'isSale'        : is_sale(item.sale)
+                }for item in product_list] 
+
+        return JsonResponse({'message' : 'SUCCESS',
+                             'data'    : {
+                                    'products' : products, 
+                                   }}, status=200)
+
```

Comment:
```suggestion

        return JsonResponse({'data' : products}, status=200)

```
데이터가 담겨가면, message는 따로 전달하지 않으셔도 될 것 같습니다 👍

Reply:
프론트의 기쁨을 위해 전달하는 메세지였는데... 이제 프론트는 웁니다 ..